src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -2293,7 +2293,7 @@ internal static bool IsAssemblyFriendOfSerialization(Assembly assembly)
             {
                 string internalsVisibleAttributeAssemblyName = internalsVisibleAttribute.AssemblyName;
 
-                if (Regex.IsMatch(internalsVisibleAttributeAssemblyName, Globals.SimpleSRSInternalsVisiblePattern) ||
+                if (internalsVisibleAttributeAssemblyName.Trim().Equals("System.Runtime.Serialization") ||
```

Comment:
Once we have https://github.com/dotnet/corefx/issues/21395, this could be done without allocating an intermediate string.
cc: @KrzysztofCwalina

Reply:
Do we have tests that cover this, including the case where there are spaces that are trimmed out?